Adding babel support to Monster Stat Box

Toiwat opened the pull request:
By substituting the hardcoded strings in dndmonster.sty with string macros, it's possible to add captions to them so that babel or polyglossia can automatically translate the field names in the moster box.

I've implemented this mostly for myself, as such, the only language I could add was Italian. Furthermore, there still isn't an official localization for 5e source material, so I went with 3.5/Pathfinder terminology where applicable.

BrianCriswell commented:
I have not used babel for much other than loading hyphenation rules. this will take me some time to look over, but it looks like it adds a nice localization hook for users. Thanks!

BrianCriswell commented:
Do the translation strings require babel to be loaded before the dnd package?

Toiwat commented:
1. As far as I know, yes; it needs to be loaded before so that other packages can pick up the language settings.

2. Sure, I'll get on to it later today once I'm on my PC.

BrianCriswell commented:
And what I mean by 2 is, "How do I switch to using your language strings?"

Toiwat commented:
I hope this is what you meant; I've added some comments in example.tex explaining how to use the language switches.

For babel, the new string macro definitions are added to \captions<language> for the given language, and caption rules are loaded when loading language definitions with \usepackage[<language>{babel}.

I've also added a base template to copy/paste to define language macros, instructions on how to use it, and a reference to it in dndstrings.sty.

Toiwat commented:
Well, I guess GitHub comment box doesn't like angle brackets... That's meant to be \captionlang where lang is the name of a language supported by babel, for example \captionsitslian, and \usepackage[lang]{babel}.

Sorry for the formatting mess.

BrianCriswell commented on lib/dndstrings-captions.sty:
```diff
@@ -0,0 +1,26 @@
+\addto\captionsitalian{%
+  \def\armorclassname{Classe Armatura}
+  \def\hitpointsname{Punti Ferita}
+  \def\speedname{Velocità}
```
LaTex does not necessarily handle accents in text well. à was garbled when I tested it. This accent should be entered as \\`a.
<https://tex.stackexchange.com/tags/accents/info>

Toiwat replied:
You're right, I forgot about that since I usually compile these with LuaLaTeX. Fixed now.
